A66-otel-stats.md

```diff
@@ -0,0 +1,325 @@
+
+To be filled
+
+## Migration from OpenCensus
+
+The following sections show the differences between the gRPC OpenCensus spec and
+the proposed gRPC OpenTelemetry spec and the mapping of metrics between the two.
+It also presents metrics present in OpenCensus spec that do not map to a metric
+in the OpenTelemetry spec at present. Two migration strategies are also proposed
+for customers who are satisfied with the stats coverage provided by the current
+OpenTelemetry spec.
+
+### Metric Schema Comparison
+
+#### Differences from gRPC OpenCensus Spec
+
+*   OpenTelemetry instrument names don’t allow ‘/’ so we use ‘.’ as the
+    separator. We also get rid of the “.io” suffix in “grpc.io” as it doesn’t
+    seem to add any value and is consistent with other names in the metrics spec
+    from OpenTelemetry.
+*   We also use this opportunity to resolve ambiguities from the gRPC OpenCensus
+    spec (detailed below).
+*   OpenTelemetry has attributes similar to tags in OpenCensus, and the
+    OpenCensus tag names already seem to match the OpenTelemetry spec - except
+    for ‘\_’ vs ‘.’ for namespaces. So we just replace the ‘\_’ with ‘.’. Note
+    that the 'client' and 'server' distinction has also been removed since it
+    does not add any benefit.
+    *   grpc_client_method -> grpc.method
+    *   grpc_client_status -> grpc.status
+    *   grpc_server_method -> grpc.method
+    *   grpc_server_status -> grpc.status
+*   Two new attributes have been added.
+    *   grpc.target - Added on client metrics
+    *   grpc.authority - Added on server metrics
+*   Latency metrics in the OpenTelemetry spec use the recommended `s` unit
+    instead of `ms`.
+
+#### Metrics with Corresponding Equivalent
+
+The following OpenCensus metrics have an equivalent in the OpenTelemetry spec
+(with the above noted differences) allowing for receivers of the telemetry data
+to join the views from the two metrics for continuity.
+
+gRPC OpenCensus                  | gRPC OpenTelemetry
+-------------------------------- | ---------------------------------------------
+grpc.io/client/started_rpcs      | grpc.client.attempt.started
+grpc.io/client/completed_rpcs    | (Derivable from grpc.client.attempt.duration)
+grpc.io/client/roundtrip_latency | grpc.client.attempt.duration
+grpc.io/server/started_rpcs      | grpc.server.call.started
+grpc.io/server/completed_rpcs    | (Derivable from grpc.server.call.duration)
+grpc.io/server/server_latency    | grpc.server.call.duration
+
+#### Metrics with Nuanced Differences
+
+Unfortunately, the implementations of the gRPC OpenCensus spec in the various
+languages do not agree on the definition of the following size metrics. Go
+records uncompressed message bytes for the OpenCensus metric, while C++ and Java
+record the compressed message bytes. The OpenTelemetry spec proposed here calls
+for recording the compressed message bytes, resulting in an equivalence between
+the metrics definitions for C++ and Java, but not for Go.
```

Comment:
it looks like the grpc-go implementation does have a compressed bytes metric (https://sourcegraph.com/github.com/grpc/grpc-go/-/blob/stats/opencensus/client_metrics.go?L39) which sounds like it would be equivalent here. is there a reason not to use that library/metric rather than [ocgrpc](https://pkg.go.dev/go.opencensus.io/plugin/ocgrpc)? we're setting up opencensus now with the intention of migrating when this work is complete, and we'd like to make sure the migration is as smooth as possible.

Reply:
Yeah, that is a newer metric recently added. If that's the metrics being used then yes it is equivalent. The one listed here is the one from the original list at https://github.com/census-instrumentation/opencensus-specs/blob/master/stats/gRPC.md